storage/innobase/fil/fil0crypt.cc

```diff
@@ -118,18 +118,28 @@ uint get_global_default_encryption_key_id_value();
 
 #define DEBUG_KEYROTATION_THROTTLING 0
 
-uint fil_get_encrypt_info_size(const uint iv_len) {
-  return ENCRYPTION_MAGIC_SIZE + 2  //length of iv
-         + 4                        // space id
-         + 2                        // offset
-         + 1                        // type
-         + 4                        // min_key_version
-         + 4                        // key_id
-         + 1                        // encryption
-         + iv_len                   // iv
-         + 4                        // encryption rotation type
-         + ENCRYPTION_KEY_LEN       // tablespace key
-         + ENCRYPTION_KEY_LEN;      // tablespace iv
+/* Return the size of encryption header used Keyring key
+encryption.
+@return length of encryption header */
+static uint fil_get_keyring_encrypt_info_size() {
```

Comment:
Mark the function constexpr

Reply:
leads to compile error:

variable declaration in a constexpr function is a C++14 extension

Reply:
Just fold const uint size ... ; return size; to return expression, and move static_assert to file scope

Reply:
Ok, then this function make any sense. Just removed it. Please check now